### ISSUE TITLE
[a11y][Violation][2.4.1 Bypass Blocks] Multiple elements with "toolbar" roles do not have unique labels 

### ISSUE THREAD

**Issue by nkn2022** [jupyterlab/jupyterlab#16905]
## Description

I am running an accessibility test suite with Jupyter using the **IBM Equal Access Accessibility Checker** tool on Chrome browser. The results look good overall, only a few things are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Listing of violations categorized based on the checkpoints. 
Full report (includes information on element locations where issue were found): 
[Accessibility_Report-Untitled.ipynb - JupyterLab.xlsx](https://github.com/user-attachments/files/17383160/Accessibility_Report-Untitled.ipynb.-.JupyterLab.xlsx)

# Category: `2.4.1 Bypass Blocks`
Ref: https://www.ibm.com/able/requirements/requirements/?version=v7_3#2_4_1

****Violation of Rule:** `aria_toolbar_label_unique`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > Multiple elements with "toolbar" roles do not have unique labels
> > **Element location:** 
> > `<jp-toolbar style="min-height: var(--jp-private-toolbar-height);" role="toolbar" aria-orientation="horizontal" orientation="horizontal" id="jp-top-bar" class="lm-Widget jp-Toolbar">`
> > **Screenshot:**
> >  <img width="468" alt="image" src="https://github.com/user-attachments/assets/74f6340b-36e0-4054-b953-1a63d1a22de4">

> > **Issue 2** 
> > Multiple elements with "toolbar" roles do not have unique labels
> > **Element location:** 
> > `<jp-toolbar style="min-height: var(--jp-private-toolbar-height);" role="toolbar" orientation="horizontal" aria-orientation="horizontal" class="lm-Widget jp-Toolbar jp-cell-menu jp-cell-toolbar">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/6572d0c1-91e1-4a30-bb27-f4edd53207d7">

## Reproduce

- Install Chrome extension for IBM accessibility checker tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open JupyterLab UI on Chrome browser. Open `notebook` file to edit.
- Run the tool and we can see the above `Violation`s reported.

Here is the screenshot of the UI scanned for the attached report:

<img width="1534" alt="image" src="https://github.com/user-attachments/assets/0e0cbc5c-cf21-4393-97fe-b566da24b58b">

## Expected behavior

Here is the suggestion from the tool: 

> **What to do**
> For each element with a "toolbar" role, either set the value of the aria-labelledby attribute to the id value of any visible text that appropriately and uniquely describes that toolbar's purpose;
> Or, use the aria-label attribute to add an invisible, unique, non-empty label to an element with the "toolbar" role.
> The following example shows two containers created with the ARIA "toolbar" role. Each container is uniquely labeled with aria-label and aria-labelledby attributes.
> 
> `<div role="toolbar" aria-label="Simple tools">`
>     ...
> `</div>`
>     
> `<div role="toolbar" aria-labelledby="ct">`
>     `<h2 id="ct">Complex tools</h2>`
>     ...
>`</div>`

## Context

- Operating System and version: MacOS Sonoma Version 14.5
- Browser and version: Chrome Version 129+
- JupyterLab version: Version 4.2.5

**Comment by krassowski** [comment]
Feel welcome to open a PR!

**Comment by nkn2022** [comment]
Sure, will do. But do I need any additional permissions. I was seeing some warnings when I tried to fork. The git fetch is not working so doubted if this is related to permissions.
> git fetch upstream
> error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
> fatal: unable to write request to remote: Broken pipe

**Comment by nkn2022** [comment]
PR is ready for review. https://github.com/jupyterlab/jupyterlab/pull/17441

**Comment by nkn2022** [comment]
There is a check error: https://github.com/jupyterlab/jupyterlab/actions/runs/14165303689/job/39677365626?pr=17441
any suggestions on how to resolve this?
